test_ros2cli/CMakeLists.txt

```diff
@@ -19,14 +19,35 @@ if(BUILD_TESTING)
   find_package(ros_testing REQUIRED)
   find_package(rmw_implementation_cmake REQUIRED)
 
-  macro(custom_test_process_output test_name config_file_name timeout delay)
+  macro(add_custom_cli_test test_name)
     # Set variables for configuring the test files.
-    set(TEST_NAME ${test_name})  # test name used in the test cases
-    set(TEST_CONFIGURATION_FILE ${config_file_name})  # config file to use
-    set(DELAY ${delay})  # Delay between the actions are launch and the ros2 command is run
+    cmake_parse_arguments(TEST "" "CONFIG_MODULE;SETUP_DELAY;TIMEOUT" "" ${ARGN})
+    # Name for test cases
+    set(TEST_NAME ${test_name})
+    # RMW implementation for test cases
+    set(TEST_RMW_IMPLEMENTATION ${rmw_implementation})
+    # Module carrying test configuration
+    if(NOT DEFINED TEST_CONFIG_MODULE)
+      set(TEST_CONFIG_MODULE ${test_name}_config)
+    endif()
+    # Setup delay for test fixture actions
+    if(NOT DEFINED TEST_SETUP_DELAY)
+      set(TEST_SETUP_DELAY 2.0)
+      if("${TEST_RMW_IMPLEMENTATION}" STREQUAL "rmw_connext_cpp")
+        # Connext startup is too slow. It needs a few extra seconds till
+        # discovery completes.
+        set(TEST_SETUP_DELAY 5.0)
+      endif()
+    endif()
+    # Timeout for test itself
+    if(NOT DEFINED TEST_TIMEOUT)
+      set(TEST_TIMEOUT 0)
```

Comment:
A timeout of `zero` doesn't sound like a good idea. Please always use an upper boundary.

Reply:
Completely agree. Changed in https://github.com/ros2/ros2cli/pull/304/commits/08d5088fda6846af14aff38915e227336e1c6819.